src/base/vesting/vesting.ts

```diff
@@ -37,50 +46,88 @@ export async function withdrawVested(
   );
   const signer = wallet.signer;
 
-  state.set("withdrawingSign");
+  state.set({ type: "withdrawingSign" });
+
+  try {
+    const tx: TransactionResponse = await contract
+      .connect(signer)
+      .withdrawVested();
 
-  const tx = await contract.connect(signer).withdrawVested();
+    state.set({ type: "withdrawing", tx });
+    await tx.wait();
+  } catch (e: any) {
+    console.log(e);
 
-  state.set("withdrawing");
-  await tx.wait();
+    state.set({ type: "error", error: e.reason });
+    return;
+  }
   session.state.refreshBalance(wallet);
-  state.set("withdrawn");
+  state.set({ type: "withdrawn" });
 }
 
 export async function getInfo(
   address: string,
   wallet: Wallet,
-): Promise<VestingInfo> {
-  const contract = new ethers.Contract(
-    address,
-    ethereumContractAbis.vesting,
-    wallet.provider,
-  );
-  const token = await contract.token();
-  const beneficiary = await contract.beneficiary();
-  const withdrawable = await contract.withdrawableBalance();
-  const withdrawn = await contract.withdrawn();
-  const total = await contract.totalVestingAmount();
-  const vestingStartTime = await contract.vestingStartTime();
-  const vestingPeriod = await contract.vestingPeriod();
-  const cliffPeriod = await contract.cliffPeriod();
-
-  const tokenContract = new ethers.Contract(
-    token,
-    ethereumContractAbis.token,
-    wallet.provider,
-  );
-  const symbol = await tokenContract.symbol();
+): Promise<VestingInfo | undefined> {
+  const contract = getVestingContract(address, wallet);
+
+  let vestingInfo:
+    | [string, string, BigNumber, BigNumber, BigNumber, string, string, string]
+    | undefined = undefined;
+  let token: string | undefined = undefined;
+
+  try {
+    token = (await contract.token()) as string;
```

Comment:
Could we move this out of the try block and just do:
```
token = await contract.token();
if(!token) {
return undefined;
}
```

Instead of the type assertion? I would find that easier to read.

Reply:
Hmm the reason I left it inside the `try` block is to catch any errors that could happen when fetching it.
I think I could leave it inside the `try` block but do the `if(!token)` guard to avoid the casting to `string`

Reply:
Something like
```js
  let token: string | undefined = undefined;

  try {
    token = await contract.token();
    if (!token) {
      return undefined;
    }
```